### PR TITLE
Fix changing workspace storage type from the overview tab

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

The PR fixes a regression when changing workspace storage type on the workspace overview tab doesn't change the workspace devfile.

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/19939

### CODE REVIEW

File: packages/dashboard-frontend/src/services/workspaceAdapter/index.ts [hunk 1]
```diff
@@ -200,7 +200,14 @@ export class WorkspaceAdapter<T extends che.Workspace | IDevWorkspace> implement
   set storageType(type: che.WorkspaceStorageType) {
     if (isWorkspaceV1(this.workspace)) {
       const attributes = typeToAttributes(type);
-      Object.assign(this.workspace.devfile.attributes, attributes);
+      if (!this.workspace.devfile.attributes) {
+        this.workspace.devfile.attributes = {};
+      }
+      if (attributes) {
+        Object.assign(this.workspace.devfile.attributes, attributes);
```
Comment (olexii4) [inline packages/dashboard-frontend/src/services/workspaceAdapter/index.ts:207]: The result of this 'Object.assign(this.workspace.devfile.attributes, attributes)' could bring a wrong storage type'.
It is better to remove 'persistVolumes' and 'asyncPersist' keys, then set needed.

Comment (akurinnoy) [reply packages/dashboard-frontend/src/services/workspaceAdapter/index.ts:207]: That makes sense, I'll provide a fix.

Comment (akurinnoy) [reply packages/dashboard-frontend/src/services/workspaceAdapter/index.ts:210]: done

**Review by olexii4 — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]
I haven't tested but changes seem to be meaningful

**Pull request merged**